### PR TITLE
RUST-395 Log skipped tests

### PR DESCRIPTION
**Pull request by karmenliang**
[RUST-395](https://jira.mongodb.org/browse/RUST-395)

This PR adds logging to test cases that are skipped and enables the `--nocapture` flag to allow the test framework to print output for skipped tests.

### CODE REVIEW

**Comment by abr-egn** [comment]
I've updated this to log skips in a way that's not captured by `cargo test` - @patrickfreed and @isabelatkinson please take a look!

**Review by isabelatkinson — APPROVED** [review]

**Review by patrickfreed — COMMENTED** [review]
functional changes LGTM--great idea with `log_uncaptured`! Only things missing are a few test names in the log messages. Also looks like this needs some merge conflicts to be fixed up

File: src/test/spec/initial_dns_seedlist_discovery.rs [hunk 1]
```diff
@@ -173,6 +179,8 @@ async fn run_test(mut test_file: TestFile) {
 
             RUNTIME.delay_for(Duration::from_millis(500)).await;
         }
+    } else {
+        log_uncaptured("skipping test due to test topology");
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/test/spec/initial_dns_seedlist_discovery.rs]: Done.

File: src/test/spec/connection_stepdown.rs [hunk 2]
```diff
@@ -31,6 +31,7 @@ async fn run_test<F: Future>(
     let client = EventClient::with_additional_options(Some(options), None, None, None).await;
 
     if !client.is_replica_set() {
+        log_uncaptured("skipping test due to not running on a replica set");
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/test/spec/connection_stepdown.rs]: Done.

File: src/test/atlas_connectivity.rs [hunk 3]
```diff
@@ -1,8 +1,11 @@
 use crate::{bson::doc, client::options::ResolverConfig, options::ClientOptions, Client};
 use bson::Document;
 
+use super::log_uncaptured;
+
 async fn run_test(uri_env_var: &str, resolver_config: Option<ResolverConfig>) {
     if std::env::var_os("MONGO_ATLAS_TESTS").is_none() {
+        log_uncaptured("skipping test due to undefined environment variable MONGO_ATLAS_TESTS");
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/test/atlas_connectivity.rs]: Done.

File: src/sdam/description/topology/test/sdam.rs [hunk 4]
```diff
@@ -643,7 +644,7 @@ async fn direct_connection() {
 
     let test_client = TestClient::new().await;
     if !test_client.is_replica_set() {
-        println!("Skipping due to non-replica set topology");
+        log_uncaptured("Skipping due to non-replica set topology");
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/sdam/description/topology/test/sdam.rs]: Done.

File: src/cmap/test/mod.rs [hunk 5]
```diff
@@ -431,7 +432,7 @@ async fn cmap_spec_tests() {
 
         let mut options = CLIENT_OPTIONS.clone();
         if options.load_balanced.unwrap_or(false) {
-            println!("skipping due to load balanced topology");
+            log_uncaptured("skipping due to load balanced topology");
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/cmap/test/mod.rs]: Done.

**Review by patrickfreed — APPROVED** [review]
LGTM! (Sorry the PR I just merged created another conflict, but besides that this should be good to go)

**Pull request merged**